### DISCUSSION TITLE
📎 Add option to vscode plugin for defining working dir

### DISCUSSION THREAD

**Discussion by Tylopilus** [rome/tools#3715] [Suggestions]
### Description

Like the possibilty within the eslint plugin one should be able to define a working dir, saved within the projects .vscode folder

This feature is very useful if you are working in a bigger project which is separated into individual modules like a monorepo. In this case if you have a setup like
* root
  * app1 (java)
  * app2 (java)
  * app3 (javascript)

it only makes sense to have the eslint/rome configs within the app3-project. If the root folder is opened in vscode though the rome.json is not recognized for working at files within app3. Only of app3 is opened directly within vscode.

This is solved by eslint by providing an option as `workingdirectory`. Their syntax is:
`"eslint.workingDirectories": [{ "directory": "app3" }]`
which then is within the .vscode folder on root-level.

This ensures that the plugin always looks for the correct configs in the app3 dir and does not fallback to defaults.

reference: https://github.com/microsoft/vscode-eslint/blob/main/history/settings_1_9_x.md

**Comment by ematipico** [comment]
The suggestion is really good! We don't have the infrastructure to support that yet, so it would take a bit of time to achieve the requested feature.
